src/labels.tsx

```diff
@@ -90,6 +93,109 @@ const renderDetailsLabel = (train: Train, prediction: Prediction | null) => {
         </div>
     );
 };
+const renderCarriageIcon = (status: OccupancyStatus, first: boolean, last: boolean) => {
+    const statusToColor: Record<typeof status, string> = {
+        EMPTY: '#008150',
+        MANY_SEATS_AVAILABLE: '#008150',
+        FEW_SEATS_AVAILABLE: '#FFC72C',
+        STANDING_ROOM_ONLY: '#FD8A03',
+        CRUSHED_STANDING_ROOM_ONLY: '#FA2D27',
+        FULL: '#FA2D27',
+        NO_DATA_AVAILABLE: 'silver',
+        NOT_ACCEPTING_PASSENGERS: 'grey',
+    };
+    return (
+        <div
+            className={`train-carriage ${first ? 'first' : ''} ${last ? 'last' : ''}`}
+            style={{
+                backgroundColor: statusToColor[status],
+            }}
+        />
+    );
+};
+
+const OccupancyStatusIcon = ({
+    occupancyStatus,
+    ...rest
+}: { occupancyStatus: OccupancyStatus } & SVGProps<SVGSVGElement>) => {
+    const occupancyStatusToIcons: Record<OccupancyStatus, number> = {
+        EMPTY: 1,
+        MANY_SEATS_AVAILABLE: 1,
+        FEW_SEATS_AVAILABLE: 2,
+        STANDING_ROOM_ONLY: 3,
+        CRUSHED_STANDING_ROOM_ONLY: 4,
+        NOT_ACCEPTING_PASSENGERS: 0,
+        NO_DATA_AVAILABLE: 0,
+        FULL: 4,
+    };
+    return (
+        <div className="occupancy-status-container">
+            {[...Array(occupancyStatusToIcons[occupancyStatus])].map((_item, i) => {
+                const hex = (255 / 4) * i;
+                const color = `rgba(${hex}, ${hex}, ${hex})`;
+                return (
+                    <PersonIcon
+                        key={i}
+                        className="occupancy-status-icon"
+                        stroke="white"
+                        strokeWidth="1px"
+                        {...rest}
+                        style={{
+                            position: 'absolute',
+                            marginLeft: i * 4,
+                            color: color,
+                            zIndex: 100 - i,
```

Comment:
these are some particularly dynamic styles to generate the occupancy status icons, hence why they're inlined. for some reason, position absolute wasn't applying correctly from the css block targeting the `occupancy-status-name` classname, so it's here.